src/main/scala/nl.knaw.dans.easy.solr4files/EasySolr4filesIndexApp.scala

```diff
@@ -112,16 +109,41 @@ trait EasySolr4filesIndexApp extends ApplicationWiring with AutoCloseable
    */
   def updateFiles(bag: Bag, ddm: DDM, filesXML: Elem): Try[BagSubmitted] = {
     (filesXML \ "file")
-      .map(FileItem(bag, ddm, _))
-      .filter(_.shouldIndex)
+      .map(getFileItem(bag, _))
```

Comment:
Since `(filesXML \ "file")` is a `NodeSeq` (or `Seq[Node]`), and `getFileItem(x, y)` returns an `Option[X]`, and given that an `Option[X]` is actually a `Seq[X]` with either 0 or 1 element, you can merge these such that the stuff below becomes easier. Then you get a `Seq[Try[FileItem]]` on this line, and then you don't need the extra unwrapping in the `map` below.

```scala
(filesXML \ "file")
  .flatMap(getFileItem(bag, _))
  .toStream
  .map(_.flatMap(createDoc(_, ddm)))
  .takeUntilFailure(...
```